### PR TITLE
migrate to github actions

### PR DESCRIPTION
**Pull request by fairclothjm**
This PR migrates this plugin from CircleCI to Github Actions. We do the following:

- remove the `.circleci` directory
  - this will cause the CircleCI Pipeline to fail until we disable the project in the CircleCI UI
- add a `.go-version` file so that we can reference it from other files
  - this is required by the `Run Tests` common workflow
- use the common `Run Tests` workflow from [vault-workflows-common](https://github.com/hashicorp/vault-workflows-common) that will
  - run the tests with `make test` and
  - build the binary with `make dev`
- reuse the common `Lint Github Actions Workflows` from [vault-workflows-common](https://github.com/hashicorp/vault-workflows-common)
  - this is a linter for Github actions themselves
  - it will detect problems with our github actions files
- reuse common "Go Checks" workflow
  - lint code with `gofumt`
  - ensure go mod is tidy
- update the `Jira Sync` action to reuse our common workflow from [vault-workflows-common](https://github.com/hashicorp/vault-workflows-common)

### CODE REVIEW

File: .github/workflows/jira.yaml [hunk 1]
```diff
@@ -25,9 +25,9 @@ jobs:
       id: preprocess
       run: |
         if [[ "${{ github.event_name }}" == "pull_request_target" ]]; then
-          echo "::set-output name=type::PR"
+          echo "type=PR" >> "$GITHUB_OUTPUT"
```
Comment (fairclothjm) [inline .github/workflows/jira.yaml:28]: `set-output` is deprecated. This was detected by actionlint.

**Review by maxcoulombe — COMMENTED** [review]
Looks good! Main comment is about DRYness.

File: .github/workflows/jira.yaml (hunk 1 above)
Comment (vinay-gopalan) [reply .github/workflows/jira.yaml:28]: Nice! I've seen the deprecation notices for set output on other repos as well, like TFVP. Good to know this is how we resolve it 😄

**Review by vinay-gopalan — APPROVED** [review]
I'm still new to setting up GH Actions, but the YAML configurations and the pipelines LGTM!

File: .github/workflows/tests.yaml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+            ~/.cache/go-build
+            ~/go/pkg/mod
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ runner.os }}-go-
+      - name: Run Tests
+        run: make test
+
+      - name: Run Test Build
+        run: make dev
```
[the remark this answers is not in the document]
Comment (fairclothjm) [reply .github/workflows/tests.yaml:43]: I think that makes sense. Thanks! PR here to DRY this up: https://github.com/hashicorp/vault-workflows-common/pull/5

File: .github/workflows/jira.yaml [hunk 3]
```diff
@@ -1,72 +1,19 @@
...
-        transition: "Pending Triage"
+    uses: hashicorp/vault-workflows-common/.github/workflows/jira.yaml
+    # assuming you use Vault to get secrets
+    # if you use GitHub secrets, use secrets.XYZ instead of steps.secrets.outputs.XYZ
+    secrets:
+      JIRA_SYNC_BASE_URL: ${{ secrets.JIRA_SYNC_BASE_URL }}
+      JIRA_SYNC_USER_EMAIL: ${{ secrets.JIRA_SYNC_USER_EMAIL }}
+      JIRA_SYNC_API_TOKEN: ${{ secrets.JIRA_SYNC_API_TOKEN }}
+    with:
+      teams-array: '["ecosystem", "foundations"]'
```
Comment (maxcoulombe) [inline .github/workflows/jira.yaml:19]: I think we own this plugin?
```suggestion
      teams-array: '["ecosystem", "applications"]'
```

File: .github/workflows/jira.yaml [hunk 4]
```diff
@@ -1,72 +1,19 @@
+name: Jira Sync
```
Comment (maxcoulombe) [inline .github/workflows/jira.yaml:1]: +1 on reusing the Jira workflow

File: .github/workflows/tests.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (maxcoulombe) [reply .github/workflows/tests.yaml:43]: Awesome thanks. If we switch to this common test workflow once its merged it solves all my questions :ship: :it:

**Review by maxcoulombe — APPROVED** [review]

File: .github/workflows/jira.yaml (hunk 3 above)
Comment (fairclothjm) [reply .github/workflows/jira.yaml:19]: oops, copy pasted without paying attention

File: .github/workflows/tests.yaml [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    runs-on: ubuntu-latest
+    outputs:
+      go-version: ${{ steps.get-go-version.outputs.go-version }}
+    steps:
+      - uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3.3.0
+      - name: Determine Go version
+        id: get-go-version
+        run: |
+          echo "Building with Go $(cat .go-version)"
+          echo "go-version=$(cat .go-version)" >> "$GITHUB_OUTPUT"
```
[the remark this answers is not in the document]
Comment (tomhjp) [reply .github/workflows/tests.yaml:20]: For my 2c, I would vote for .go-version file as it specifies the patch version too, so it's a plus for reproducibility. Also if we use go.mod file's go version then we are overloading the meaning of a number which already has fairly subtle and [independently evolving meaning](https://github.com/golang/go/issues/57001).

[the remark this answers is not in the document]
Comment (fairclothjm) [reply .github/workflows/tests.yaml:20]: @tomhjp I think those are convincing enough reasons to outweigh the convenience of overloading go.mod's go version.

[the remark this answers is not in the document]
Comment (fairclothjm) [reply .github/workflows/tests.yaml:20]: I went ahead and opened a PR on the common repo to use the .go-version file and move actionlint to that repo: https://github.com/hashicorp/vault-workflows-common/pull/7

File: .github/workflows/tests.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (fairclothjm) [reply .github/workflows/tests.yaml]: Updated to use common test workflow in https://github.com/hashicorp/vault-plugin-auth-alicloud/pull/63/commits/21f93b4080bc018e0cbf181820b560398cf67622

File: .github/workflows/jira.yaml [hunk 6]
```diff
@@ -1,72 +1,19 @@
...
-        issue: ${{ steps.search.outputs.issue }}
-        transition: Closed
-
-    - name: Reopen ticket
-      if: github.event.action == 'reopened' && steps.search.outputs.issue
-      uses: atlassian/gajira-transition@v2.0.1
-      with:
-        issue: ${{ steps.search.outputs.issue }}
-        transition: "Pending Triage"
+    uses: hashicorp/vault-workflows-common/.github/workflows/jira.yaml@f49ac86e61bf70e595c6b7d90b73cfb8378e0a16
```
Comment (fairclothjm) [inline .github/workflows/jira.yaml:11]: should be:
```suggestion
uses: hashicorp/vault-workflows-common/.github/workflows/jira.yaml@main
```

Comment (fairclothjm) [reply .github/workflows/jira.yaml:11]: fixed in https://github.com/hashicorp/vault-plugin-auth-alicloud/pull/68

**Pull request merged**